src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -245,12 +245,7 @@ impl<'a> Dependency for (&'a PackageId, Stage) {
         let (id, stage) = *self;
         let pkg = packages.iter().find(|p| p.package_id() == id).unwrap();
         let deps = resolve.deps(id).into_iter().flat_map(|a| a)
-                          .filter(|dep| *dep != id)
-                          .map(|dep| {
-                              (dep, pkg.dependencies().iter().find(|d| {
-                                  d.name() == dep.name()
-                              }).unwrap())
-                          });
```

Comment:
Could this logic be left here actually? It's still used in multiple places below, and the call to `find` could be replaced with `filter` and then the cases below could use the iterator if necessary.